data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDB.java

```diff
@@ -76,4 +83,81 @@ public Environment getEnvironment(
 
     return environmentMapper.toEnvironment(environmentDDBRecord);
   }
+
+  @Override
+  public Environment createEnvironmentTargetVersion(final Environment environment)
+      throws StorageException, EnvironmentExistsException {
+
+    final EnvironmentTargetVersionDDBRecord environmentTargetVersionDDBRecord =
+        environmentMapper.toEnvironmentTargetVersionDDBRecord(environment);
+
+    try {
+      dynamoDBMapper.save(environmentTargetVersionDDBRecord);
+    } catch (final AmazonServiceException e) {
+      throw new StorageException(
+          String.format(
+              "Could not save record with environment id %s", environment.getEnvironmentId()),
+          e);
+    }
+    return environmentMapper.toEnvironment(environmentTargetVersionDDBRecord);
+  }
+
+  @Override
+  public Environment getEnvironmentTargetVersion(String environmentId)
+      throws StorageException, EnvironmentNotFoundException {
+
+    final EnvironmentTargetVersionDDBRecord environmentTargetVersionDDBRecord;
+    try {
+      environmentTargetVersionDDBRecord =
+          dynamoDBMapper.load(EnvironmentTargetVersionDDBRecord.withHashKey(environmentId));
+    } catch (final AmazonServiceException e) {
+      throw new StorageException(
+          String.format("Could not load record with environment id %s", environmentId), e);
+    }
+
+    if (environmentTargetVersionDDBRecord == null) {
+      throw new EnvironmentNotFoundException(
+          String.format("Could not find environment with id %s", environmentId));
+    }
+
+    return environmentMapper.toEnvironment(environmentTargetVersionDDBRecord);
+  }
+
+  @Override
+  public List<String> listClusters() throws StorageException {
+    try {
+      List<EnvironmentTargetVersionDDBRecord> scanResult =
+          dynamoDBMapper.scan(
```

Comment:
Can't we get this from scanning the cluster GSI instead of scanning through the entire EnvironmentTargetVersion table?

Reply:
oh did that in the query and not here? silly